private/objectiveWrap.m

```diff
@@ -58,6 +59,10 @@
                     case 'log-posterior'          , varargout = {-J,-G(I)};
                     case 'negative log-posterior' , varargout = { J, G(I)};
                 end
+                if any(any(isnan(G))) || any(any(isinf(G)))
```

Comment:
Would use [isfinite](https://de.mathworks.com/help/matlab/ref/isfinite.html) instead; this checks for `nan` and `inf` simultaneously. Same for Hessian and gradient below.